### ISSUE TITLE
Parser fails on TemplateLiteral in object property in VExpressionContainer

### ISSUE THREAD

**Issue by levchak0910** [vuejs/vue-eslint-parser#187]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I'm using [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue).
- [X] I'm sure the problem is a parser problem. (If you are not sure, search for the issue in [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) repo and open the issue in [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) repo if there is no solution.
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.15.0

### What version of `eslint-plugin-vue` and ` vue-eslint-parser` are you using?

- vue-eslint-parser@9.1.1
- eslint-plugin-vue@9.10.0

### What did you do?

Configuration = fork of eslint-plugin-vue

Trying to get a correct parsed result

```vue
<template>
  <div :class="{`some`: true}" /> <!-- backticks! -->
</template>
```

### What did you expect to happen?

I expect to see a child `ObjectExpression` with `Property[name=some][type=TemplateLiteral]` in `VExpressionContainer` 

### What actually happened?

I see `null` in an expression of `VExpressionContainer` 

here is a screen of [astexplorer](https://astexplorer.net/)
version of vue-eslint-parser is 7.6.0 there, but the parsing result is the same for 9.1.1
![image](https://user-images.githubusercontent.com/37511777/230787692-1f76c8b0-5637-4d6f-a814-88744b7fd86e.png)

### Link to Minimal Reproducible Example

```js
module.exports = {
  meta: {
    type: 'problem',
    fixable: null,
    schema: [],
  },
  /** @param {RuleContext} context */
  create(context) {
    return utils.defineTemplateBodyVisitor(context, {
      VAttribute(node) {
        console.log(node.value.expression) // = null
      },
    })
  },
}
```

### Additional comments

let me know if a reproducible repository is anyway needed

**Comment by levchak0910** [comment]
Seems I am wrong
It is not possible to create object with keys as template literal
```js
const b = {
  `some`: true,
}
```
`> Uncaught SyntaxError: Unexpected template string`
